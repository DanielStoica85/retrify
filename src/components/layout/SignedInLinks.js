import React from 'react';
import { NavLink, NavItem, Badge, Button } from 'reactstrap';
import { connect } from 'react-redux';
import { logout } from '../../actions/auth';

export const SignedInLinks = props => {
    return (
        <React.Fragment>
            <NavItem>
                <NavLink href="/retro">New Retro</NavLink>
            </NavItem>
            <NavItem>
                <NavLink href="/">
                    <Badge color="danger" pill>
                        D.S.
                    </Badge>
                </NavLink>
            </NavItem>
            <NavItem>
                <Button onClick={props.logout}>Log Out</Button>
            </NavItem>
        </React.Fragment>
    );
};

const mapDispatchToProps = dispatch => {
    return {
        logout: () => dispatch(logout())
    };
};

export default connect(
    null,
    mapDispatchToProps
)(SignedInLinks);
