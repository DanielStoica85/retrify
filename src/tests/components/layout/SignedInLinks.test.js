import React from 'react';
import { shallow } from 'enzyme';
import SignedInLinks from '../../../components/layout/SignedInLinks';

describe('SignedInLinks component', () => {
    it('should render correctly', () => {
        const component = shallow(<SignedInLinks />);
        expect(component).toMatchSnapshot();
    });
});
