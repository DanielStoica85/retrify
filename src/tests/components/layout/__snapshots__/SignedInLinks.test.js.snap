// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedInLinks component should render correctly 1`] = `
<Fragment>
  <NavItem
    tag="li"
  >
    <NavLink
      href="/retro"
      tag="a"
    >
      New Retro
    </NavLink>
  </NavItem>
  <NavItem
    tag="li"
  >
    <NavLink
      href="/"
      tag="a"
    >
      Log Out
    </NavLink>
  </NavItem>
  <NavItem
    tag="li"
  >
    <NavLink
      href="/"
      tag="a"
    >
      <Badge
        color="danger"
        pill={true}
        tag="span"
      >
        D.S.
      </Badge>
    </NavLink>
  </NavItem>
</Fragment>
`;
